CMakeLists.txt

```diff
@@ -17,18 +17,17 @@ find_package(Eigen3 REQUIRED)
 # TODO(dhood): enable python support once ported to ROS 2
 # catkin_python_setup()
 
-include_directories(include
-    ${rclcpp_INCLUDE_DIRS}
-    ${sensor_msgs_INCLUDE_DIRS}
-    ${tf2_INCLUDE_DIRS}
-    ${Eigen3_INCLUDE_DIRS}
-)
-
 add_library(laser_geometry SHARED src/laser_geometry.cpp)
-target_link_libraries(laser_geometry
-    ${rclcpp_LIBRARIES}
-    ${sensor_msgs_LIBRARIES}
-    ${tf2_LIBRARIES}
+target_include_directories(laser_geometry
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+  ${Eigen3_INCLUDE_DIRS}
+)
+ament_target_dependencies(laser_geometry
+  "rclcpp"
```

Comment:
@Karsten1987 do these need to be quoted?

Reply:
I guess they don’t have to be quoted since every variable is interpreted as a string (I might be wrong here)
But we do that in other parts of the ROS2 code base as well, e.g. in rclcpp: https://github.com/ros2/rclcpp/blob/master/rclcpp/CMakeLists.txt#L112

Reply:
That's fair, and I'll go ahead and merge this. I do think that quoting some things but not others (for example the arguments to find_package listed earlier in the file) might confuse people. In general I prefer only adding required characters, so that people don't wonder what they really need and what is optional.